src/Server/src/Abstractions/Model/ClientScope.cs

```diff
@@ -2,11 +2,61 @@
 
 namespace IdOps.Model
 {
-    public class ClientScope
+    public class ClientScope : IEquatable<ClientScope>
```

Comment:
This is wrong. 
Just comparing the Id does not respect the intend of the interface. Also, why do we even need the interface in first place?

I guess we just need it because of the hashset on AllowedGrantTypes. But there we hav e another porblem

Reply:
The interface has no intend, the meaning of the equitable depends on how you design the model and this is an entity with and id which means that if the id is equal than you are talking about the same entity.

This has nothing to do with `AllowedGrantTypes`, it's needed to can create a set of distinct scopes.